### PR TITLE
feature(addon-google-analytics) Make tracking of specific storybook core events configurable

### PR DESCRIPTION
**Pull request by alexanderjeurissen**
## What I did

This PR is based on the work in https://github.com/storybookjs/storybook/pull/11336
and based on the issue discussion with @shilman 

It further improves the analytics add-on in the following ways:

1. Improve resilience of the code, by adding typescript typing for ReactGA initialisation / options.
2. Add `events` add-on parameter that enables users to `opt-in `/`opt-out` of tracking specific storybook core events (this opens up expanding the number of events that we can log to GA without being noisy as it can be opt-in).

## How to test

- Is this testable with Jest or Chromatic screenshots?
  - This is probably testable with Jest, although I could not find existing test coverage for this addon. If I'm missing something please advise.
- Does this need a new example in the kitchen sink apps?
  - I don't think so, as we currently don't have analytics in the kitchen sink apps.
- Does this need an update to the documentation?
  - Yes, I updated the README to reflect the configuration change

### CODE REVIEW

File: addons/google-analytics/README.md [hunk 1]
```diff
@@ -20,9 +26,83 @@ module.exports = {
...
+
+```js
+import { addParameters } from '@storybook/client-api';
+import { STORY_CHANGED } from '@storybook/core-events';
+
+addParameters({
+  analytics: {
+    reactGAId: 'UA-000000-01',
+    reactGAOptions: {},
+    events: {
```
Comment (alexanderjeurissen) [inline addons/google-analytics/README.md:70]: not sure if `events` is a very descriptive configuration parameter for defining what events we want to track.

I'm open to suggestions on this.

File: addons/google-analytics/README.md [hunk 2]
```diff
@@ -20,9 +26,83 @@ module.exports = {
...
+      [STORY_ERRORED]: ({ title }) => title.includes('Expecting a React element from the story')
+    }
+  },
+});
+```
+In this case, the function returns true if the title of the error matches the provided string.
+The function return value in turn, determines if the `STORY_ERRORED` event is tracked or not.
```
Comment (alexanderjeurissen) [inline addons/google-analytics/README.md:108]: Copy can probably use some tweaks to be more concise and clear.

**Comment by MichaelArestad** [comment]
Google Analytics is no longer part of this repository. Thanks for the PR. 🙌

**Comment by shilman** [comment]
new repo https://github.com/storybookjs/addon-google-analytics

**Pull request closed without merging**